### ISSUE TITLE
Early Stopping Iter does not work

### ISSUE THREAD

**Issue by thomasATbayer** [catboost/catboost#2567]
Problem: Early Stopping with od_type = "Iter" does not work. According to the Manual it is supposed to work without a validation set.
But training does not stop even if Metric does not improve anymore

catboost version: 1.2.2

Here is the code to replicate:
from itertools import product

import numpy as np
from catboost import CatBoostClassifier, Pool, cv
from sklearn.metrics import roc_auc_score
from sklearn.datasets import make_blobs

from matplotlib import pyplot
from pandas import DataFrame

# generate 2d classification dataset
X,y = make_blobs(n_samples=1000, centers=2, n_features=1000)

# Split dataset into train and test sets
train_size = int(0.8 * len(X))
X_train, X_test = X[:train_size], X[train_size:]
y_train, y_test = y[:train_size], y[train_size:]

# Create CatBoost dataset
train_pool = Pool(X_train, y_train)

#train a model using log loss

loss = "Logloss"
model_log_loss = CatBoostClassifier(od_type = "Iter",od_wait = 10, loss_function=loss, eval_metric="MCC", verbose=2)
test_pool = Pool(X_test)

model_log_loss.fit(train_pool)

**Comment by ek-ak** [comment]
Hello! Thank you for you feedback.
Please specify where in the manual did you find that the conversion detector should work without a validation set? It should not.
